tensorrt_llm/_torch/modules/linear.py

```diff
@@ -1789,6 +1797,7 @@ def __init__(
         allreduce_strategy: AllReduceStrategy = AllReduceStrategy.AUTO,
         force_dynamic_quantization: bool = False,
         use_cute_dsl_blockscaling_mm: bool = False,
```

Comment:
I think we should have a better design to tune gemms between trtllm-gen/cutlass/cutedsl.

And do not leak the information of backend in the Linear interface if possible.

Linear is a logic block, choosing which one of the backend shall happen automatically based on the input data type, GPU, and kernel performance.

Reply:
It's better to add cute_dsl kernel in firstly and then replace gradually. Because the baseline (cutlass impls) perf is so good. We need to do step-by-step optimization and then make sure which case could use cute_dsl kernel in default.